partner_affiliate/tests/test_partner_affiliate.py

```diff
@@ -0,0 +1,84 @@
+from odoo.tests import common
+
+
+class TestPartnerAffiliate(common.TransactionCase):
+
+    def setUp(self):
+        super(TestPartnerAffiliate, self).setUp()
+
+        self.first_parent = self.env['res.partner'].create({
+            'name': 'MyFirstParentForTheAffiliate',
+            'type': 'contact',
+            'is_company': True,
+            'street': 'first parent street',
+            'street2': 'number 99',
+            'zip': 123,
+            'city': 'Test City',
+        })
+
+        self.second_parent = self.env['res.partner'].create({
+            'name': 'MySecondParentForTheAffiliate',
+            'type': 'contact',
+            'is_company': True,
+            'street': 'second parent street',
+            'street2': 'number 44',
+            'zip': 999,
+            'city': 'Test City',
+        })
+
+    def test_change_parent_from_a_new_affiliate(self):
+        new_affiliate = self.env['res.partner'].create({
+            'name': 'MyTestAffiliate',
+            'is_company': True,
+            'parent_id': self.first_parent.id,
+            'type': 'affiliate',
+            'street': 'affiliate street',
+            'street2': 'number 11',
+            'zip': 567,
+            'city': 'Test City',
+            'email': 'myAffiliate@test.com',
+        })
+
+        # Checks for data integrity in affiliate and his parent.
+        self.assertTrue(new_affiliate, "The new affiliate have been created.")
+
+        self.assertEquals(new_affiliate.type, 'affiliate',
+                          "Check type must be 'affiliate'")
+        self.assertEquals(new_affiliate.parent_id.id, self.first_parent.id,
+                          "Must be child of the parent defined in the setup")
+        self.assertEquals(new_affiliate.street, "affiliate street",
+                          "The street have been correctly set.")
+        self.assertEquals(self.first_parent.street, "first parent street",
+                          "The parent continues with his original street")
+
+        # Change the parent of the affiliate for the second one in the set-up.
+        new_affiliate.write({'parent_id': self.second_parent.id})
+        new_affiliate.onchange_parent_id()
```

Comment:
This is wrong. An onchange should be applied on a temporal object, otherwise the object is already saved and the onchange wasn't applied. What is doing that onchange?

Reply:
Sorry. I'm wrong. The test is right. 
You have an existing partner and you want to modify the partner and apply the onchange. 
I mixing up concepts, as I was using onchange with new in some other tests to check what happens in a temporal object, but this is not the case of your test.  
This is the way to do it, although you could use 
new_affiliate.parent_id = self.second_parent.id
This is the new API way instead of the current old API version.
Just do this change, run the tests and I will merge it. 
Good job!

Reply:
Done. Thanks for the tip.